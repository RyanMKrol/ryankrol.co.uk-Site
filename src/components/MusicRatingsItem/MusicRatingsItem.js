import React, { Component } from 'react';

/**
 * A component representing all information about a "listens" item
 */
class MusicRatingsItem extends Component {
  /**
   * @class
   * @param {object} props arguments to class
   */
  constructor(props) {
    super();

    this.highlights = props.highlights;
    this.artist = props.artist;
    this.mood = props.mood;
    this.date = props.date;
    this.rating = props.rating;
    this.title = props.title;

    if (
      !this.highlights
      || !this.artist
      || !this.mood
      || !this.date
      || !this.rating
      || !this.title
    ) {
      throw new Error('Could not create MusicRatingsItem');
    }
  }

  /**
   * Render method
   *
   * @returns {React.Component} JSX MusicRatingsItem component
   */
  render() {
    return (
      <div>
        <span className="highlighted">Highlights:</span>
        <span>{this.highlights}</span>
        <br />
        <span className="highlighted">Artist:</span>
        <span>{this.artist}</span>
        <br />
        <span className="highlighted">Mood:</span>
        <span>{this.mood}</span>
        <br />
        <span className="highlighted">Date:</span>
        <span>{this.date}</span>
        <br />
        <span className="highlighted">Rating:</span>
        <span>{this.rating}</span>
        <br />
        <span className="highlighted">Title:</span>
        <span>{this.title}</span>
        <br />
      </div>
    );
  }
}

export default MusicRatingsItem;
