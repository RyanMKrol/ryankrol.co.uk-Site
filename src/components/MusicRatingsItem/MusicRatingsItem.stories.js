import React from 'react';

import DEFAULT_MOCK from './__mocks__';

export default {
  title: 'Components/MusicRatingsItem',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the MusicRatingsItem
 */
export const normal = () => DEFAULT_MOCK;
