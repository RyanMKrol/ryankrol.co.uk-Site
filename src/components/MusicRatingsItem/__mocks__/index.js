import MusicRatingsItem from '../MusicRatingsItem';

const DEFAULT_MOCK = (
  <MusicRatingsItem
    highlights={'Selah, Follow God'}
    artist={'Kanye West'}
    mood={'Vibin'}
    date={'16-05-2021'}
    rating={71}
    title={'JESUS IS KING'}
  />
);

export default DEFAULT_MOCK;
