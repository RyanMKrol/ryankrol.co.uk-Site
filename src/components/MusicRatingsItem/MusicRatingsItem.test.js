import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import MusicRatingsItem from './MusicRatingsItem';

import DEFAULT_MOCK from './__mocks__/MusicRatingsItemMocks';

it('renders MusicRatingsItem without issue', () => {
  const listensItem = shallow(DEFAULT_MOCK);
  expect(toJson(listensItem)).toMatchSnapshot();
});

it('renders MusicRatingsItem without throwing', () => {
  expect(() => mount(DEFAULT_MOCK)).not.toThrow();
});

it('MusicRatingsItem throws without highlights', () => {
  expect(() => mount(<MusicRatingsItem />)).toThrow();
});
it('MusicRatingsItem throws without artist', () => {
  expect(() => mount(<MusicRatingsItem highlights={'test-highlights'} />)).toThrow();
});
it('MusicRatingsItem throws without mood', () => {
  expect(() => mount(<MusicRatingsItem highlights={'test-highlights'} artist={'test-artist'} />)).toThrow();
});
it('MusicRatingsItem throws without date', () => {
  expect(() => mount(
      <MusicRatingsItem highlights={'test-highlights'} artist={'test-artist'} mood={'test-mood'} />,
  )).toThrow();
});
it('MusicRatingsItem throws without rating', () => {
  expect(() => mount(
      <MusicRatingsItem
        highlights={'test-highlights'}
        artist={'test-artist'}
        mood={'test-mood'}
        date={'test-data'}
      />,
  )).toThrow();
});
it('MusicRatingsItem throws without title', () => {
  expect(() => mount(
      <MusicRatingsItem
        highlights={'test-highlights'}
        artist={'test-artist'}
        mood={'test-mood'}
        date={'test-data'}
        title={'test-title'}
      />,
  )).toThrow();
});
