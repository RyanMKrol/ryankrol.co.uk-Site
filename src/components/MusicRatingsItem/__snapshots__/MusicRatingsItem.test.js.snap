// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MusicRatingsItem without issue 1`] = `
<div>
  <span
    className="highlighted"
  >
    Highlights:
  </span>
  <span>
    Selah, Follow God
  </span>
  <br />
  <span
    className="highlighted"
  >
    Artist:
  </span>
  <span>
    Kanye West
  </span>
  <br />
  <span
    className="highlighted"
  >
    Mood:
  </span>
  <span>
    Vibin
  </span>
  <br />
  <span
    className="highlighted"
  >
    Date:
  </span>
  <span>
    16-05-2021
  </span>
  <br />
  <span
    className="highlighted"
  >
    Rating:
  </span>
  <span>
    71
  </span>
  <br />
  <span
    className="highlighted"
  >
    Title:
  </span>
  <span>
    JESUS IS KING
  </span>
  <br />
</div>
`;
